osu.Game/Overlays/MusicController.cs

```diff
@@ -181,6 +181,37 @@ private bool next(bool instant = false)
             return false;
         }
 
+        private bool restartTrack(out bool isRestart)
```

Comment:
The name of this function is kinda misleading, I'm not good with namings but maybe rename to `startOrRestartTrack`?

Reply:
If there's an `Or` in a method name, it's a surefire indicator that the method is doing too much.

Reply:
Yeah again, I'm not good with namings.